aws-samples/i-love-my-local-farmer
# flochaz/DeliveryApi

First version of Delivery project infrastructure.

### Unit tests

```
cd Projects/Delivery/cdk
mvn package
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< com.ilmlf.delivery:api >-----------------------
[INFO] Building api 0.1
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.2:check (validate) @ api ---
[INFO] You have 0 Checkstyle violations.
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ api ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ api ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ api ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/chazalf/amazon/EMEASASpecialistBuilders/ILoveMyLocalFarmer/i-love-my-local-farmer/Projects/Delivery/cdk/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ api ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /Users/chazalf/amazon/EMEASASpecialistBuilders/ILoveMyLocalFarmer/i-love-my-local-farmer/Projects/Delivery/cdk/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ api ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.ilmlf.delivery.DeliveryAppTest
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#noProviders for further details.
pwd
Bundling asset DeliveryProject-Api/CreateSlots/Code/Stage...
Script executed successfully
Bundling asset DeliveryProject-Api/CreateSlots/Code/Stage...
Script executed successfully
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 30.482 s - in com.ilmlf.delivery.DeliveryAppTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ api ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  39.148 s
[INFO] Finished at: 2021-05-26T17:16:36+02:00
[INFO] ------------------------------------------------------------------------

```

### Integ tests

Manually testing through API Gateway

Method Execution /farm/{farm-id}/slots - POST - Method Test
farm-id: 2
Body:
```
{
      "slots": [
          {
              "numDeliveries": "2",
              "from": "2020-05-28T10:00:00",
              "to": "2020-05-28T11:00:00"
          }
       ]
}
```
Response:

Request: /farm/2/slots
Status: 200
Latency: 2317 ms
Response Body
```
Slot data (1) was saved successfully
```
Response Headers
```
{"X-Amzn-Trace-Id":"Root=1-60af5230-5e7736f650c6f774409af9b7;Sampled=0"}
```


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
